### DISCUSSION TITLE
How to support chinese characters in windows virtual enviroment ?

### DISCUSSION THREAD

**Discussion by shinny-hongyan** [community/community#26166] [Actions]
<p>I run pytest in workflow,and the chinese characters display in unicode,just like this :</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d8c356e67547b7c713b2c437c09d60423c52aabd.jpeg" data-download-href="/uploads/short-url/uVzE51cqh4Dgtf0Gtl1wT1e4kkR.jpg?dl=1" title="errinfo.jpg" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/59548734/181095895-4228f71e-ec24-4592-a471-117300bf1a35.jpg" alt="errinfo.jpg" data-base62-sha1="uVzE51cqh4Dgtf0Gtl1wT1e4kkR" width="690" height="153" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d8c356e67547b7c713b2c437c09d60423c52aabd_2_690x153.jpg, https://aws1.discourse-cdn.com/github/optimized/2X/d/d8c356e67547b7c713b2c437c09d60423c52aabd_2_1035x229.jpg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/d/d8c356e67547b7c713b2c437c09d60423c52aabd_2_1380x306.jpg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d8c356e67547b7c713b2c437c09d60423c52aabd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">errinfo.jpg</span><span class="informations">1509×335 72 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So, how to support chinese characters in GH actions Terminal ?</p>

**Comment by Yanjingzhu** [comment]
<p> Have you set  <strong># -*- coding: utf-8 -*-</strong> in the pytest file? </p>
<p>I tried to output chinese characters in powershell scripts directly, it worked as expected. To narrow down your issue, whether it related to python or github hosted windows virtual enviroment, could you please share a sample repository here for us to reproduce and investigate this issue further? </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/602f3510e9fa2207ce03f30f10a50bc5f6dbc824.png" data-download-href="/uploads/short-url/dISYEDDbR88oDusC0O1PXzeBQbi.png?dl=1" title="ps chinese.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095897-8b29cebc-3bef-450a-85ea-5f753af2bff5.png" alt="ps chinese.png" data-base62-sha1="dISYEDDbR88oDusC0O1PXzeBQbi" width="690" height="328" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/602f3510e9fa2207ce03f30f10a50bc5f6dbc824_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">ps chinese.png</span><span class="informations">867×413 9.66 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by shinny-hongyan** [comment]
<p>I try your solution but it does not work , here is my repo:<a href="https://github.com/shinnytech/tqsdk-python.git" rel="nofollow noopener">https://github.com/shinnytech/tqsdk-python.git</a></p>
<p>and here is my actions which otput unicode instead of chinese character:<a href="https://github.com/shinnytech/tqsdk-python/runs/489953096?check_suite_focus=true" rel="nofollow noopener">https://github.com/shinnytech/tqsdk-python/runs/489953096?check_suite_focus=true</a></p>
<p>tks!</p>

**Comment by shinny-hongyan** [comment, marked as answer]
<p>well, I have solved this problem, and I fixed it by add:</p>
<pre><code>env:
      PYTHONIOENCODING: "utf-8"
</code></pre>
<p>here are the tips: <a href="https://github.com/pytest-dev/pytest/issues/3640" rel="nofollow noopener">https://github.com/pytest-dev/pytest/issues/3640</a></p>
